juju/juju
# Extend port APIs on provider interfaces to take a source range attribute

This is a largely mechanical change to alter the port related methods on the Environ and Instance interfaces. We introduce a new network.IngressRule struct which embeds a PortRange but also includes a SourceRange which defines the allowed source CIDRs of the incoming packets.

Open/ClosePorts() and Ports() are the affected methods. Various providers eg MAAS are just no-ops, so the changes for those a just to the method definitions.

For Azure and GCE, the methods are changed but the implementation is left as is for now ie any passed in source ranges are ignored. Juju doesn't actually make use of source ranges yet. A number of shared unit tests in livetests.go only deal with port ranges. Helper methods are used to convert between port ranges and ingress rules until all providers can be updated.

For AWS and Openstack, the source ranges are mapped to security group rules. Openstack's security groups on allow for one source attribute so one ingress rule is mapped to many security groups and vice versa.

The worker/firewaller is updated to work with the new interfaces but does not yet fill out any source ranges. So there's no functional change introduced by this PR.

As  a drive by, some unused code was deleted in the network package.

QA
bootstrap aws and openstack
deploy and expose an appliction
check that the expected ports are opened for 0.0.0.0/0 according to rules in the machine level security groups
